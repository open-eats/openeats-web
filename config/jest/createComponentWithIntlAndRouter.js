import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom'

const createComponentWithIntl = (children, props = { locale: 'en' }) => {
  return renderer.create(
    <IntlProvider {...props}>
      <MemoryRouter>
        { children }
      </MemoryRouter>
    </IntlProvider>
  );
};

export default createComponentWithIntl;