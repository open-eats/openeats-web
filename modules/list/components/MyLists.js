import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import { injectIntl, defineMessages } from 'react-intl'

const MyLists = ({title, lists, intl}) => {
  const messages = defineMessages({
    no_lists: {
      id: 'my_lists.no_lists',
      description: 'No Lists to display',
      defaultMessage: 'No Lists to display',
    },
    new_list: {
      id: 'my_lists.new_list',
      description: 'Create a New List!',
      defaultMessage: 'Create a New List!',
    },
  });

  if (lists === null || lists.length === 0) {
    return (
      <div className="grocery-lists">
      <ul className="list-group">
        <p className="list-group-item disabled">
          { title }
        </p>
        <p className="list-group-item disabled">
          { intl.formatMessage(messages.no_lists) }
        </p>
      </ul>
    </div>
    );
  }

  let items = lists.map(function(item) {
    let link = '/list/' + item.slug;
    return (
      <NavLink
        to={ link }
        className="list-group-item"
        activeClassName="active"
        key={ item.id }
      >
        <span className="badge">{ item.item_count }</span>
        { item.title }
      </NavLink>
    );
  });

  return (
    <div className="grocery-lists">
      <ul className="list-group">
        <p className="list-group-item disabled">
          { title }
        </p>
        { items }
        <NavLink exact={ true } to={ '/list/' } className="list-group-item">
          { intl.formatMessage(messages.new_list) }
        </NavLink>
      </ul>
    </div>
  );
};

MyLists.propTypes = {
  title: PropTypes.string.isRequired,
  intl: PropTypes.object.isRequired,
  lists: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    item_count: PropTypes.number.isRequired
  }).isRequired).isRequired,
};

export default injectIntl(MyLists)
