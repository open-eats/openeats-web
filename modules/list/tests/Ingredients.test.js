import React from 'react';
import MyLists from '../components/MyLists';
import createComponentWithIntlAndRouter from '../../../config/jest/createComponentWithIntlAndRouter';

import listData from './listData';

test('MyLists component test', () => {
  const component = createComponentWithIntlAndRouter(
    <MyLists title="My Lists" lists={ listData }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('MyLists component test w/ no lists', () => {
  const component = createComponentWithIntlAndRouter(
    <MyLists title="My Lists" lists={ [] }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
