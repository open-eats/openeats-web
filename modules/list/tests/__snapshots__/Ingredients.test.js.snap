// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLists component test 1`] = `
<div
  className="grocery-lists"
>
  <ul
    className="list-group"
  >
    <p
      className="list-group-item disabled"
    >
      My Lists
    </p>
    <a
      aria-current={false}
      className="list-group-item"
      href="/list/1"
      onClick={[Function]}
      style={undefined}
    >
      <span
        className="badge"
      >
        3
      </span>
      The first test
    </a>
    <a
      aria-current={false}
      className="list-group-item"
      href="/list/sdfsdf"
      onClick={[Function]}
      style={undefined}
    >
      <span
        className="badge"
      >
        4
      </span>
      another test
    </a>
    <a
      aria-current={false}
      className="list-group-item"
      href="/list/test"
      onClick={[Function]}
      style={undefined}
    >
      <span
        className="badge"
      >
        0
      </span>
      test
    </a>
    <a
      aria-current={false}
      className="list-group-item"
      href="/list/"
      onClick={[Function]}
      style={undefined}
    >
      Create a New List!
    </a>
  </ul>
</div>
`;

exports[`MyLists component test w/ no lists 1`] = `
<div
  className="grocery-lists"
>
  <ul
    className="list-group"
  >
    <p
      className="list-group-item disabled"
    >
      My Lists
    </p>
    <p
      className="list-group-item disabled"
    >
      No Lists to display
    </p>
  </ul>
</div>
`;
